### DISCUSSION TITLE
ListBoxItem's behaviors?

### DISCUSSION THREAD

**Discussion by XDgl233** [AvaloniaUI/Avalonia#12736] [Q&A] [answered]
Hi all! I am currently working on a custom listbox class, and with that, I want to achieve some very simple UI behaviors like this:
The user selects the item and then taps the confirm button. The confirmed item will show a check icon on it showing this item is being marked.

But when I implemented these, I noticed that whenever I scrolled the main window, the marked list box items that were out of the screen and not visible would "jump" to other places. I bound the data context to the view model, and I noticed that bound property in vm was totally intact, while this seemed like a totally visual glitch. I opened up the Avalonia debug program, and noticed  the list box item will lost the entire datacontext when they are out of the screen. Can I assume that, the list box items are actively destroyed and created when needed? And what approach I should take to make my ui logic work? Thx :)

**Comment by timunie** [comment]
Avalonia version? If 11.0, please file a minimal sample to reproduce.

**Reply by XDgl233** [reply]
Hi Tim! Thanks for your help :)
https://github.com/XDgl233/ListItemScrollingTest/tree/main

This is the minimal sample I tried and tested in my working environment.
The project pops some items in the list, and you can select one of the item and hit the button select, then if you scroll the item list, I hope that sample can reproduce the problem I had.

**Reply by timunie** [reply, marked as answer]
Why so complicated? The only valid option I see in your usecase is to disable virtualization which is a bad idea. Imo better to have an Item-ViewModel which holds confirmation-status.

**Reply by XDgl233** [reply]
I figured it out and used list item view models as you suggested. Works like a charm :)

I thought the confirmation should be part of the view's business, and created this mess, but really this should be the business of the view model :)
